### PR TITLE
Added network function module

### PR DESCRIPTION
**Pull request by kukulkarni1**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1175
    - [ ] txe markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/NetworkFunction/custom/Get-AzNetworkFunctionTrafficCollectorPolicy.ps1 [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    try {
+        $outBuffer = $null
+        if ($PSBoundParameters.TryGetValue('OutBuffer', [ref]$outBuffer)) {
+            $PSBoundParameters['OutBuffer'] = 1
+        }
+        $parameterSet = $PSCmdlet.ParameterSetName
+        $mapping = @{
+            Get = 'Az.NetworkFunction.private\Get-AzNetworkFunctionTrafficCollectorPolicy_Get';
+            GetViaIdentity = 'Az.NetworkFunction.private\Get-AzNetworkFunctionTrafficCollectorPolicy_GetViaIdentity';
+            List = 'Az.NetworkFunction.private\Get-AzNetworkFunctionTrafficCollectorPolicy_List';
```
Comment (wyunchi-ms) [inline src/NetworkFunction/custom/Get-AzNetworkFunctionTrafficCollectorPolicy.ps1:123]: You can just call the `internal` cmdlet after process the `PSBoundParameters`. [Here](https://github.com/Azure/azure-powershell/blob/567d576adbeb042838debd69a9a26842c16ddc8c/src/Aks/Aks.Autorest/custom/Get-AzAksVersion.ps1#L112) is an example.

File: src/NetworkFunction/custom/Az.NetworkFunction.custom.psm1 [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# region Generated 
+  # Load the private module dll
+  $null = Import-Module -PassThru -Name (Join-Path $PSScriptRoot '..\bin\Az.NetworkFunction.private.dll')
+
+  # Load the internal module
+  $internalModulePath = Join-Path $PSScriptRoot '..\internal\Az.NetworkFunction.internal.psm1'
+  if(Test-Path $internalModulePath) {
+    $null = Import-Module -Name $internalModulePath
+  }
+
```
Comment (wyunchi-ms) [inline src/NetworkFunction/custom/Az.NetworkFunction.custom.psm1:10]: No need to commit this file.

File: src/NetworkFunction/custom/Models/AzureTrafficCollector.PowerShell.cs [hunk 3]
```diff
@@ -0,0 +1,216 @@
+// Code generated by Microsoft (R) AutoRest Code Generator (autorest: 3.8.0, generator: @autorest/powershell@3.0.481)
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+namespace Microsoft.Azure.PowerShell.Cmdlets.NetworkFunction.Models
+{
+    using Microsoft.Azure.PowerShell.Cmdlets.NetworkFunction.Runtime.PowerShell;
+
+    /// <summary>Azure Traffic Collector resource.</summary>
+    [System.ComponentModel.TypeConverter(typeof(AzureTrafficCollectorTypeConverter))]
+    public partial class NetworkFunction
```
Comment (wyunchi-ms) [inline src/NetworkFunction/custom/Models/AzureTrafficCollector.PowerShell.cs:10]: The class name should be `AzureTrafficCollector`. And it's better to use the models generated under `generated/Models`.

File: src/NetworkFunction/custom/Get-AzNetworkFunctionTrafficCollector.ps1 [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+            $PSBoundParameters['OutBuffer'] = 1
+        }
+        $parameterSet = $PSCmdlet.ParameterSetName
+        $mapping = @{
+            Get = 'Az.NetworkFunction.private\Get-AzNetworkFunctionTrafficCollector_Get';
+            GetViaIdentity = 'Az.NetworkFunction.private\Get-AzNetworkFunctionTrafficCollector_GetViaIdentity';
+            List = 'Az.NetworkFunction.private\Get-AzNetworkFunctionTrafficCollector_List';
+        }
+        if (('Get', 'List') -contains $parameterSet -and -not $PSBoundParameters.ContainsKey('SubscriptionId')) {
+            $PSBoundParameters['SubscriptionId'] = (Get-AzContext).Subscription.Id
```
Comment (wyunchi-ms) [inline src/NetworkFunction/custom/Get-AzNetworkFunctionTrafficCollector.ps1:127]: I don't see special custom logic here, I think you can just use generated cmdlet.

File: src/NetworkFunction/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+title: NetworkFunction
+subject-prefix: $(service-name)
+identity-correction-for-post: true
+resourcegroup-append: true
+nested-object-to-string: true
+
+directive:
+  - where:
+      variant: ^Create$|^CreateViaIdentity$|^CreateViaIdentityExpanded$|^Update$|^UpdateViaIdentity$
+    remove: true
```
Comment (wyunchi-ms) [inline src/NetworkFunction/README.md:42]: Change `remove: true` to `hide: true` will generate the cmdlets under `internal` folder.

Comment (kukulkarni1) [reply src/NetworkFunction/README.md]: Changed to hide: true

File: src/NetworkFunction/custom/Az.NetworkFunction.custom.psm1 (hunk 2 above)
Comment (kukulkarni1) [reply src/NetworkFunction/custom/Az.NetworkFunction.custom.psm1]: Removed this file

File: src/NetworkFunction/custom/Get-AzNetworkFunctionTrafficCollector.ps1 (hunk 4 above)
Comment (kukulkarni1) [reply src/NetworkFunction/custom/Get-AzNetworkFunctionTrafficCollector.ps1]: Removed custom scripts

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**